changelog/unreleased/pull-3419

```diff
@@ -0,0 +1,18 @@
+Enhancement: Use config file modes to derive new file/dir modes
+
+Previously restic would always create new files and directories on
+local/sftp repositories with hardcoded permissions of 0700/0600
+(rwx/rw). We now use the current read permissions of the "config" file to
+derive which modes new files and directories should be created with.
```

Comment:
That is not really correct. E.g. a permission of 0444 will never allow anyone to read/write files in the repository. It's just the question whether the files should be group accessible or not.